metpy/io/pandas_to_netcdf.py

```diff
@@ -0,0 +1,174 @@
+
+    # Attach variable-specific metadata
+    _assign_metadata(dataset_final, column_units, standard_names, long_names)
+
+    # Attach dataset-specific metadata
+    if dataset_type:
+        dataset_final.attrs['featureType'] = dataset_type
+    else:
+        log.warning('No dataset type provided - netCDF will not have appropriate metadata'
+                    'for a DSG dataset.')
+    if dataset_type:
+        dataset_final[sampling_var].attrs['cf_role'] = dataset_type.lower() + '_id'
+    dataset_final['samplingIndex'].attrs['instance_dimension'] = 'samples'
+
+    # Determine mode to write to netCDF
+    write_mode = 'w'
+    if path.exists(path_to_save):
+        # Eventually switch to 'a' to allow appending and delete error
+        raise ValueError('File already exists - please delete and run again')
+
+    # Check if netCDF4 is installed to see how many unlimited dimensions we can use
+    # Need conditional import for checking due to Python 2
+    try:
+        from importlib.util import find_spec
+        check_netcdf4 = find_spec('netCDF4')
+    except ImportError:
+        from imp import find_module
+        check_netcdf4 = find_module('netCDF4')
+
+    # Make sure path is a string to allow netCDF4 to be used - needed for tests to pass
+    path_to_save = str(path_to_save)
+
+    if check_netcdf4 is not None:
+        unlimited_dimensions = ['samples', 'observations']
+    else:
+        # Due to xarray's fallback to scipy if netCDF4-python is not installed
+        # only one dimension can be unlimited. This may cause issues for users
+        log.warning('NetCDF4 not installed - saving as a netCDF3 file with only the'
+                    'observations dimension as unlimited. If netCDF4 or multiple'
+                    'dimensions are desired, run `pip install netCDF4`')
+        unlimited_dimensions = ['observations']
+
+    # Convert to netCDF
+    dataset_final.to_netcdf(path=path_to_save, mode=write_mode, format=netcdf_format,
+                            unlimited_dims=unlimited_dimensions, compute=True)
+
+
+def _assign_metadata(dataset, units_dict, standard_names_dict, long_names_dict):
+    if units_dict is not None:
+        final_column_units = {}
+        final_column_units['samples'] = ''
+        final_column_units['observations'] = ''
+        final_column_units['samplingIndex'] = ''
+        final_column_units.update(units_dict)
+        for var in dataset.variables:
+            dataset[var].attrs['units'] = final_column_units[var]
+    if standard_names_dict is not None:
+        final_std_names = {}
+        final_std_names['samples'] = ''
+        final_std_names['observations'] = ''
+        final_std_names['samplingIndex'] = ''
+        final_std_names.update(standard_names_dict)
+        for var in dataset.variables:
+            dataset[var].attrs['standard_name'] = final_std_names[var]
+    else:
+        log.warning('No standard names provided - netCDF file will not be CF compliant.')
```

Comment:
I don't think CF actually ever requires you to use standard names.

Reply:
Yeah you're right on this. Says it's optional in v1.7.